force-app/main/default/classes/RD2_ScheduleService.cls

```diff
@@ -95,6 +106,9 @@ public without sharing class RD2_ScheduleService {
         schedule.Campaign__c = rd.npe03__Recurring_Donation_Campaign__c;
 
         schedule.InstallmentAmount__c = rd.npe03__Amount__c;
+        if(nextPaymentAmount != null) {
```

Comment:
I think we end up with nextPaymentAmount on both schedules for a 1st and 15th cadence. Is that what we want?

Reply:
On second thought, this shouldn't hurt anything as long as we clear both of them after the one-time payment is paid. Makes the logic a bit cleaner, no need to figure out which schedule to set 🤷